tests/integration/test_task_remote_mgr.py

```diff
@@ -0,0 +1,104 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from cylc.flow.task_remote_mgr import (
+    REMOTE_FILE_INSTALL_DONE,
+    REMOTE_FILE_INSTALL_FAILED
+)
+
+
+async def test_remote_tidy(
```

Comment:
I don't love this test. Suggestions for the streamlining of it would be gratefully accepted.

Reply:
Looks ok to me TBH. How do you prevent remote-tidy from actually trying to run on the remote platforms?

Reply:
I don't. I just don't worry about it when they fail because that's after the chunk of code I'm trying to test

I considered making the hosts `uuids` to ensure that they remote clean would fail reliably. I got strange results when I had `foo` in my `~/.ssh/config`

Reply:
Maybe just monkeypatch it out?

Reply:
Good idea, now done